src/components/buttons/AddToPlayListBtn.js

```diff
@@ -0,0 +1,148 @@
+    const [showPlayListModal, setShowPlayListModal] = useState(false);
+    return (
+        <>
+            <button
+                onMouseEnter={() => setShowDescription(true)}
+                onMouseLeave={() => setShowDescription(false)}
+                onClick={() => setShowPlayListModal(true)}
+                className="static"
+            >
+                <i className="fas fa-list-ul fa-lg"></i>
+                {showDescription && (
+                    <div className="hidden sm:block absolute bg-gray-300 p-1 rounded text-black">
+                        Add To Play List
+                    </div>
+                )}
+            </button>
+            {showPlayListModal && (
+                <PlayListModal
+                    setShowPlayListModal={setShowPlayListModal}
+                    video={video}
+                />
+            )}
+        </>
+    );
+}
+
+function PlayListModal({ setShowPlayListModal, video }) {
+    const [newPlayListName, setNewPlayListName] = useState('');
+
+    const {
+        state: { playLists },
+        dispatch,
+    } = useData();
+
+    function addVideoToPlayList(playList) {
+        toast.success(`Video removed from ${playList.name}`);
+        dispatch({
+            type: 'REMOVE_VIDEO_FROM_PLAYLIST',
+            payload: { name: playList.name, video: video },
+        });
+    }
+
+    function removeVideoFromPlayList(playList) {
+        toast.success(`Video added to ${playList.name}`);
+        dispatch({
+            type: 'ADD_VIDEO_TO_PLAYLIST',
+            payload: { name: playList.name, video: video },
+        });
+    }
+
+    async function addNewPlayList() {
+        if (playLists.find((list) => list.name === newPlayListName)) {
+            toast.error('Play List Already Exists, try using another name');
+            return;
+        }
+        setNewPlayListName('');
+        dispatch({
+            type: 'CREATE_NEW_PLAYLIST',
+            payload: { name: newPlayListName, videos: [] },
+        });
+    }
+
+    function isVideoInPlayList(playList) {
+        if (
+            playList.videos.find((item) => item.videoId === video.videoId) !==
+            undefined
+        ) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    async function handleCheckBoxChange(playList) {
+        try {
+            if (isVideoInPlayList(playList)) {
+                addVideoToPlayList(playList);
+            } else {
+                removeVideoFromPlayList(playList);
+            }
+        } catch (error) {
+            console.log(error);
+        }
+    }
+
+    return (
+        <div
+            id="modal"
+            className=" flex justify-center items-center flex-col
+            fixed w-screen h-screen bg-gray-800 
+            bg-opacity-50 top-0 right-0"
+            onClick={(e) =>
+                e.target.id === 'modal' ? setShowPlayListModal(false) : null
+            }
```

Comment:
Why onClick on div ? See if this can be altered.
Although you have set setShowPlayListModal(false) to X in modal to close modal. 
So, I think not required here

Reply:
I have done this so that the modal closes when you click anywhere outside the sidebar.